packages/machine-extractor/src/sky/skyConfigNode.ts

```diff
@@ -18,11 +22,42 @@ export type SkyConfigNodeReturn = WithValueNodes<{
 }> &
   Pick<ObjectPropertyInfo, 'node'>;
 
+const StringLiteralOrExpression = unionType([
+  StringLiteral,
+  maybeTsAsExpression(
+    maybeIdentifierTo(
+      createParser({
+        babelMatcher: t.isMemberExpression,
+        parseNode: (node, context): StringLiteralNode => {
+          if (!context.getNodeSource || !context.getEnvVariable) {
+            throw new Error("Couldn't find API key in any of the env files");
+          }
+          // Let's find the last part of the expression, e.g. `API_KEY` in `process.env.API_KEY`
+          const envVariableName = context
+            .getNodeSource(node)
+            .match(/(?<=\.)(\w+)(?!.*\.)/);
```

Comment:
instead of looking into this as a string you could do smth like this:
```ts
const envVariableName = t.buildMatchMemberExpression('process.env')(node.object) && t.isIdentifier(path.node.property) ? node.property.name : null
```

Reply:
Since you also want to handle `import.meta` then
```ts
const envVariableName = (t.isMetaProperty(node.object) || t.buildMatchMemberExpression('process.env')(node.object)) && t.isIdentifier(node.property)
 ? node.property.name
  : null
```

Reply:
Please recheck if that works. I tried to quickly confirm that using the https://astexplorer.net/ and I think it's fine but better to be safe than sorry

Reply:
It works great; I tested it before adding it in. Using both `process.env.KEY` and `import.meta.KEY`